Sources/Hub/Hub.swift

```diff
@@ -190,9 +190,32 @@ public class LanguageModelConfigurationFromHub {
     ) async throws -> Configurations {
         // Note tokenizerConfig may be nil (does not exist in all models)
         let modelConfig = try hubApi.configuration(fileURL: modelFolder.appending(path: "config.json"))
+        // First try to get the tokenizer_config.json
         let tokenizerConfig = try? hubApi.configuration(fileURL: modelFolder.appending(path: "tokenizer_config.json"))
+        // Check for chat_template.json, which contains the preferred chat template for vision language models
+        if let chatTemplateConfig = try? hubApi.configuration(fileURL: modelFolder.appending(path: "chat_template.json")) {
```

Comment:
Technically, this is not the same algorithm used in transformers. IIRC, if we instantiate a tokenizer from a repo where the tokenizer has a chat template and a different `chat_template.json`, the template from the tokenizer will still be used. However, if we instantiate a `processor`, then the `chat_template.json` will be used.

I'm willing to diverge from this behaviour, given that:
- Chat template divergence should not be expected (it's possibly a mistake if both templates differ)
- Processor and tokenizer templates should be synced at some point.
- There is no `processor` abstraction in swift-transformers.

cc @Rocketknight1 in case I'm missing some weird edge case.